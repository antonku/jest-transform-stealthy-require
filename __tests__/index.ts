const transformer = require('../lib');

describe('jest-stealthy-require-transform', () => {

  it('should transform stealthyRequire calls into jest.isolateModules() when stealthyRequire identifier has an initializer', () => {
    const source = `
      var stealthyRequire = require('stealthy-require');
      var request = stealthyRequire(require.cache, function () {
        return require('request');
        },
        function () {
          require('tough-cookie');
      }, module);
    `;
    const transformed = transformer.process(source);
    expect(transformed).toMatchSnapshot();
  });

  it('should transform stealthyRequire calls into jest.isolateModules() when stealthyRequire identifier does not have an initializer', () => {
    const source = `
      var stealthyRequire;
      stealthyRequire = require('stealthy-require');
      var request = stealthyRequire(require.cache, function () {
        return require('request');
        },
        function () {
          require('tough-cookie');
      }, module);
    `;
    const transformed = transformer.process(source);
    expect(transformed).toMatchSnapshot();
  });

});
