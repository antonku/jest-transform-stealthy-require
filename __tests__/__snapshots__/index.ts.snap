// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-stealthy-require-transform should transform stealthyRequire calls into jest.isolateModules() when stealthyRequire identifier does not have an initializer 1`] = `
"var stealthyRequire;
stealthyRequire = require('stealthy-require');
var request;
jest.isolateModules(function () {
    request = require('request');
});"
`;

exports[`jest-stealthy-require-transform should transform stealthyRequire calls into jest.isolateModules() when stealthyRequire identifier has an initializer 1`] = `
"var stealthyRequire = require('stealthy-require');
var request;
jest.isolateModules(function () {
    request = require('request');
});"
`;
